infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -68,12 +77,40 @@ public MachineResolver(
   public InternalMachineConfig resolve() throws InfrastructureException {
     InternalMachineConfig machineConfig =
         new InternalMachineConfig(
-            null, toServers(containerEndpoints), null, null, toWorkspaceVolumes(cheContainer));
+            null,
+            toServers(containerEndpoints),
+            null,
+            toMachineAttributes(pluginId, wsAttributes),
+            toWorkspaceVolumes(cheContainer));
 
     normalizeMemory(container, machineConfig);
     return machineConfig;
   }
 
+  private Map<String, String> toMachineAttributes(
```

Comment:
I think it would be better to rewrite it in more `Single Responsibility Principle` style

```java

  private Map<String, String> toMachineAttributes(String pluginId, Map<String, String> wsAttributes) {
   Map<String, String> attributes = new HashMap<>();

    Optional<String> pluginAlias = findPluginAlias(pluginId, wsAttributes);
    pluginAlias.ifPresent(s -> attributes.put(DEVFILE_COMPONENT_ALIAS_ATTRIBUTE, s));

    return attributes;
  }

  private Optional<String> findPluginAlias(String pluginId, Map<String, String> wsAttributes) {

    List<String> aliases = new ArrayList<>();

    String pluginComponentAliases = wsAttributes.get(PLUGINS_COMPONENTS_ALIASES_WORKSPACE_ATTRIBUTE);
    if (!isNullOrEmpty(pluginComponentAliases)) {
      aliases.addAll(asList(pluginComponentAliases.split(",")));
    }

    String editorComponentAlias = wsAttributes.get(EDITOR_COMPONENT_ALIAS_WORKSPACE_ATTRIBUTE);
    if (!isNullOrEmpty(editorComponentAlias)) {
      aliases.add(editorComponentAlias);
    }

    if (aliases.isEmpty()) {
      return Optional.empty();
    }

    return aliases
          .stream()
          .map(value -> value.split("="))
          .filter(arr -> arr[0].equals(pluginId))
          .map(arr -> arr[1])
          .findAny();
  }

```

Reply:
makes sense